Generator/source/CMakeLists.txt

```diff
@@ -1,58 +1,104 @@
+project(CircuitsGenerator 
+  VERSION 0.1.0
+  DESCRIPTION "Programm for circuits dataset generation"
+  HOMEPAGE_URL https://github.com/vvzunin/CAD_CombinationalCircuits
+  LANGUAGES CXX
+)
 
-include(ExternalProject)
+# Add path for .cmake files. Needed for "include" directive.
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-SET(GCC_COVERAGE_COMPILE_FLAGS "-fprofile-arcs -ftest-coverage")
-SET(GCC_COVERAGE_LINK_FLAGS    "-lgcov")
+# Big section with some variables to define
+# Define variables for coverege
+set(GCC_COVERAGE_COMPILE_FLAGS "-fprofile-arcs -ftest-coverage")
+set(GCC_COVERAGE_LINK_FLAGS    "-lgcov")
+# Define main cmake project settings
 set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_VERBOSE_MAKEFILE ON)
+#set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_VERBOSE_MAKEFILE OFF)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_INIT} -std=c++11 -fPIC")
 set(CMAKE_CXX_FLAGS_RELEASE "-O0")
 set(CMAKE_BUILD_TYPE Debug)
-
+# Define folder for external libraries
 set(EXTERNAL_INSTALL_LOCATION ${CMAKE_BINARY_DIR}/external)
+# Define name of final file.
+set(MAIN_TARGET circuitgen)
+set(MAIN_LIBRARY cgen)
+# Needed to build library.
+set(BUILD_LIBRARY True)
 
-option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+OPTION(GENERATOR_DEBUG "Extended debug printing" OFF)
 
-#ExternalProject_Add(googletest
-#  GIT_REPOSITORY https://github.com/google/googletest
-#  GIT_TAG main
-#  SOURCE_DIR "${CMAKE_BINARY_DIR}/googletest-src"
-#  BINARY_DIR "${CMAKE_BINARY_DIR}/googletest-build" 
-#  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTERNAL_INSTALL_LOCATION}
-#)
+# Configuration file.
+configure_file(config.h.in config.h)
 
-#add_subdirectory (${CMAKE_BINARY_DIR}/external/lib/cmake/GTest)
+option(BUILD_SHARED_PROJECT_LIBS "Build using shared libraries" ON)
 
+include(ExternalProject)
 include(FetchContent)
 
-FetchContent_Declare(json
-  GIT_REPOSITORY https://github.com/nlohmann/json.git
-  GIT_TAG v3.7.3)
-
-FetchContent_GetProperties(json)
-if(NOT json_POPULATED)
-  FetchContent_Populate(json)
-  add_subdirectory(${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
+# Get json
+FetchContent_Declare(
+  json
+  GIT_REPOSITORY https://github.com/nlohmann/json
+  GIT_TAG v3.11.3
+)
+FetchContent_MakeAvailable(json)
 
+# Get googletest
 FetchContent_Declare(
   googletest
   URL https://github.com/google/googletest/archive/03597a01ee50ed33e9dfd640b249b4be3799d395.zip
 )
 
-
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
 FetchContent_MakeAvailable(googletest)
 
+# The compilation options.
+# add_compile_options(-g3 -Wall -Wignored-qualifiers -DYYERROR_VERBOSE)
 
+# Add all subdirectories for the project.
+add_subdirectory(AuxiliaryMethods)
+add_subdirectory(circuits)
+add_subdirectory(DataBase)
+add_subdirectory(filesTools)
+add_subdirectory(generators)
+add_subdirectory(graph)
+add_subdirectory(optimization_utils)
+add_subdirectory(reliability)
+add_subdirectory(settings)
+add_subdirectory(thread_pool)
```

Comment:
As I can see there is a kind of mess for subdirectory names. Some of them are "CamelCase", some are "lowerscore_and_separated". It would be good to select some common practice.